File: sesion3-lab/spring-rate-limit/rate-limit-server/src/main/java/com/example/demo/web/ApiController.java
package com.example.demo.web;

import java.time.Instant;
import java.util.Map;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api")
public class ApiController {

    @GetMapping("/datos-sensibles")
    public Map<String, Object> datosSensibles() {
        return Map.of(
                "message", "Datos servidos correctamente (dentro del limite de peticiones).",
                "timestamp", Instant.now().toString());
    }
}

File: sesion3-lab/aspnet-rate-limit/ratelimit-demo/Controllers/ApiController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.RateLimiting;

namespace RateLimitDemo.Controllers;

[ApiController]
[Route("api")]
public class ApiController : ControllerBase
{
    [HttpGet("datos-sensibles")]
    [EnableRateLimiting("api-policy")]
    public IActionResult ObtenerDatos()
    {
        return Ok(new
        {
            message = "Datos servidos correctamente (dentro del limite de peticiones).",
            timestamp = DateTimeOffset.UtcNow
        });
    }

    // Endpoint sin rate limiting, para comparar.
    [HttpGet("publico")]
    public IActionResult Publico()
    {
        return Ok(new
        {
            message = "Endpoint sin rate limiting: responde siempre.",
            timestamp = DateTimeOffset.UtcNow
        });
    }
}
